### ISSUE TITLE
Add rules to clarify valid aria-invalid use

### ISSUE THREAD

**Issue by scottaohara** [w3c/html-aria#366]
`aria-invalid` can have the following values:
* `true` - to indicate the value entered by the user is invalid
* `false` - default state, no issues with validity 
* `grammar` - a grammatical error was found
* `spelling` - spelling error found

ARIA 1.2 is deprecating the attribute from being global as the `true`, and in some ways `false`, values are specifically suited for form controls.  `grammar` and `spelling` however, are still necessary for rich text editing, and though the attribute is being deprecated globally, conformance checkers should not outright flag its use on non-form controls to be incorrect.

So, in regards to this spec, we would need to call out that `aria-invalid=true|false` should be allowed on the form controls where this makes sense, but otherwise (specifically `aria-invalid=true`) not be allowed on other elements.  But `aria-invalid=grammar|spelling` would still be allowed on any HTML element that otherwise allows for global `aria-*` attributes.

related to https://github.com/w3c/aria/issues/989

**Comment by scottaohara** [comment]
Requires issue 391 (linked above) be merged to create the section of the spec to include this information.
